OSDOCS-1861: Add discussion of multi-network policy

jboxman opened the pull request:
- https://issues.redhat.com/browse/OSDOCS-1861

~So this rebased off of https://github.com/openshift/openshift-docs/pull/32708 and all those changes appear here as well. Ideally, 32708 is merged first.~ The following content is specific to this PR:

Previews
- [Configuring multi-network policy](https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-multi-network-policy.html)

(So mostly `modules/nw-multi-network-policy-*.adoc` files.)

jboxman commented:
Hi @s1061123, can you take a look at this multi-network policy content when you have a moment? Thanks!

weliang1 commented on modules/nw-networkpolicy-create.adoc:
```diff
@@ -1,59 +1,120 @@
 ----
-$ oc create -f <policy-name>.yaml -n <project>
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
```
Should be apiVersion: k8s.cni.cncf.io/v1beta1?

weliang1 commented on modules/nw-networkpolicy-create.adoc:
```diff
@@ -1,59 +1,120 @@
 ----
+ifndef::multi[]
 kind: NetworkPolicy
 apiVersion: networking.k8s.io/v1
```
Should it be apiVersion: k8s.cni.cncf.io/v1beta1?

jboxman replied:
@weliang1, yes, I've added a conditional for it because this file is included in 2 different assemblies.

weliang1 commented on modules/nw-networkpolicy-delete.adoc:
```diff
@@ -1,25 +1,41 @@
 [source,terminal]
 ----
-$ oc delete networkpolicy <policy-name>
+$ oc delete networkpolicy <policy_name>
```
oc delete multi-networkpolicy

weliang1 commented on modules/nw-networkpolicy-delete.adoc:
```diff
@@ -1,25 +1,41 @@
+.Example output
+[source,text]
+----
+networkpolicy.networking.k8s.io/allow-same-namespace deleted
```
Change to "multinetworkpolicy.k8s.cni.cncf.io "ingress-allow-diff-podselector-with-same-namespaceselector" deleted
"

jboxman replied:
@weliang1, would this be correct for creating a multi-network policy?

`multinetworkpolicy.k8s.cni.cncf.io/default-deny created`

jboxman commented:
@weliang1, I've linked to additional resources for delete, edit, ect. I expect that a user understands that these require using `multi-networkpolicy` instead of `networkpolicy` in commands.

```
* xref:../../networking/network_policy/about-network-policy.adoc#about-network-policy[About network policy]
* xref:../../networking/network_policy/viewing-network-policy.adoc#viewing-network-policy[Viewing a network policy]
* xref:../../networking/network_policy/editing-network-policy.adoc#editing-network-policy[Editing a network policy]
* xref:../../networking/network_policy/deleting-network-policy.adoc#deleting-network-policy[Deleting a network policy]
```

Although I can try to make it clear in the introduction to multi-network policy.

mikemckiernan commented on modules/nw-multi-network-policy-enable.adoc:
```diff
@@ -0,0 +1,40 @@
+
+.Procedure
+
+. Create a file with the following YAML:
```
Nothing wrong with what you have, but I'll share an idea.  You could be prescriptive and suggest creating a file named `multi-network-policy.yaml` in this step and then reference the same file name in the next.  That would eliminate the somewhat superfluous suggestion to substitute the real file name.  I imagine that customers understand they can name files whatever they want.

mikemckiernan commented on modules/nw-multi-network-policy-enable.adoc:
```diff
@@ -0,0 +1,40 @@
+  useMultiNetworkPolicy: true
+----
+
+. To enable multi-network policy, enter the following command. Replace `<name>` with the name of the YAML file you created in the previous step.
```
Maybe "Configure the cluster to enable multi-network policy:"

I believe you stated a preference for "To <goal>, enter..." but that ends up with sentence structure that isn't parallel with the first step.  I also suspect that you can drop the "Replace..." sentence even if you don't choose to be prescriptive.

mikemckiernan commented on modules/nw-networkpolicy-create.adoc:
```diff
@@ -6,6 +6,11 @@
+
+[id="nw-networkpolicy-create_{context}"]
 
 [id="nw-networkpolicy-create_{context}"]
```
Just an eye check...did you mean to add this again?

mikemckiernan reviewed (approved):
Looks good.

weliang1 commented:
Not see "oc get multi-networkpolicy" in https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/viewing-network-policy.html#viewing-network-policy
Not see "oc edit multi-networkpolicy" in https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/editing-network-policy.html#editing-network-policy
Not see "oc delete multi-networkpolicy" in https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/deleting-network-policy.html#deleting-network-policy

weliang1 commented:
> @weliang1 multi-networkpolicy and Kuberneres networkpolicy are different feature, so I believe we should have different pages for each feature, otherwise user may get confused about multi-networkpolicy and Kubernetes networkpolicy.

@jboxman I agree @s1061123 suggestion.

jboxman commented:
@weliang1 @s1061123 

I've included all the multi-network policy procedures. I use conditional expressions to use `multi-networkpolicy` in place of `networkpolicy`.

https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-multi-network-policy.html

weliang1 commented:
> @weliang1 @s1061123
> 
> I've included all the multi-network policy procedures. I use conditional expressions to use `multi-networkpolicy` in place of `networkpolicy`.
> 
> https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-multi-network-policy.html

@jboxman should you include "oc edit multi-networkpolicy" and "oc delete multi-networkpolicy" in https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-multi-network-policy.html?

weliang1 commented:
/LGTM

jboxman commented:
@weliang1 thank you!

@mikemckiernan, I changed the structure some and added conditionals everywhere, can you take a look when you have a minute? Thanks!

jboxman commented:
/cherry-pick enterprise-4.8
